### PR TITLE
Climber subsystem

### PR DESCRIPTION
**Pull request by Levercpu**
We set up some code that allows for the robot to automtically climb and balance based on the pitch of the navx gyro #best.

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -59,4 +59,21 @@ public final class Constants {
     public static final double FRONT_LEFT_ABS_ENCODER_ZERO = 339.5;
     public static final double BACK_LEFT_ABS_ENCODER_ZERO = 18.72;
     public static final double FRONT_RIGHT_ABS_ENCODER_ZERO = 133.3;
+
+    public static final double ArmSeperationDistance = 5.00;
```
Comment (bengold22) [inline src/main/java/frc/robot/Constants.java]: Why does your PR have INTAKE constants? Please remove any constant not related to the code in this PR.

File: src/main/java/frc/robot/commands/StaticClimb.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    private final Climber climber;
+    private double startTime;
+
+    public StaticClimb(Climber climber) {
+        this.climber = climber;
+        addRequirements(climber);
+    }
+    @Override
+    public void initialize() {
+        System.out.println("Im Climbing");
```
Comment (bengold22) [inline src/main/java/frc/robot/commands/StaticClimb.java]: Remove the System.out.println

File: src/main/java/frc/robot/commands/StaticClimb.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        } else if (-climber.getGyroPitch()>Constants.CLIMBER_BALANCE_THRESH) {
+            climber.balanceright(Constants.CLIMBER_BALANCE_LOW_SPEED);
+        } else {
+            climber.raise(false);
+        }
+        
+    }
+    @Override
+    public boolean isFinished() {
+        if (Timer.getFPGATimestamp()-startTime>10) {
```
Comment (bengold22) [inline src/main/java/frc/robot/commands/StaticClimb.java]: Move 10 to constants (something like CLIMB_TIMEOUT_S)

Comment (bengold22) [reply src/main/java/frc/robot/commands/StaticClimb.java]: Also lets reduce it now to 4

File: src/main/java/frc/robot/commands/StaticClimb.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    }
+    @Override
+    public void execute() {
+        if (climber.getGyroPitch() > Constants.CLIMBER_BALANCE_THRESH) {
+            climber.balanceleft(Constants.CLIMBER_BALANCE_LOW_SPEED);
+        } else if (-climber.getGyroPitch()>Constants.CLIMBER_BALANCE_THRESH) {
+            climber.balanceright(Constants.CLIMBER_BALANCE_LOW_SPEED);
+        } else {
+            climber.raise(false);
```
Comment (bengold22) [inline src/main/java/frc/robot/commands/StaticClimb.java]: I think this is supposed to be true, not false

File: src/main/java/frc/robot/subsystems/Intake.java
Comment (bengold22) [inline src/main/java/frc/robot/subsystems/Intake.java:1]: Please delete this file, it looks to me like this was experimental code (it's using PWM)

File: src/main/java/frc/robot/subsystems/Outtake.java
Comment (bengold22) [inline src/main/java/frc/robot/subsystems/Outtake.java:1]: Delete this file, same an Intake.java

File: src/main/java/frc/robot/utils/IntakeState.java
Comment (bengold22) [inline src/main/java/frc/robot/utils/IntakeState.java:1]: Delete this file

File: src/main/java/frc/robot/utils/OuttakeState.java
Comment (bengold22) [inline src/main/java/frc/robot/utils/OuttakeState.java:1]: Delete this file

**Review by bengold22 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -5,28 +5,28 @@
...
-import frc.robot.subsystems.Ramp;
 import frc.robot.utils.smartshuffleboard.SmartShuffleboard;
-import edu.wpi.first.wpilibj.XboxController;
 
 
 public class RobotContainer {
   private final Joystick joyleft = new Joystick(Constants.LEFT_JOYSICK_ID);
   private final Joystick joyright = new Joystick(Constants.RIGHT_JOYSTICK_ID);
   private final SwerveDrivetrain drivetrain;
+  private Climber climber = null;
```
Comment (noahheller) [inline src/main/java/frc/robot/RobotContainer.java]: there is no need to initialize climber to null. Consider removing the null and making Climber final.

File: src/main/java/frc/robot/RobotContainer.java [hunk 6]
```diff
@@ -56,13 +58,13 @@ public RobotContainer() {
   public void putShuffleboardCommands() {
     SmartShuffleboard.putCommand("Ramp", "SetArmPID400", new RampMove(ramp, 400));
     SmartShuffleboard.putCommand("Ramp", "SetArmPID500", new RampMove(ramp, 500));
-
+    SmartShuffleboard.putCommand("Climber", "Climb", new StaticClimb(climber));
+    SmartShuffleboard.putCommand("TEST","WheelAlign",new WheelAlign(drivetrain));
```
Comment (noahheller) [inline src/main/java/frc/robot/RobotContainer.java]: Should be in wrapped in an if(CLIMBER_DEBUG)

File: src/main/java/frc/robot/subsystems/Climber.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        SparkMax1.set(0);
+        SparkMax2.set(speed);
+    }
+    public void stop(){
+        SparkMax1.set(0);
+        SparkMax2.set(0);
+    }
+    @Override
+    public void periodic() {
+        SmartShuffleboard.put("Climber", "Climber State", "Pitch", getGyroPitch());
```
Comment (noahheller) [inline src/main/java/frc/robot/subsystems/Climber.java]: another If debug

File: src/main/java/frc/robot/Constants.java [hunk 8]
```diff
@@ -59,4 +59,19 @@ public final class Constants {
...
```
Comment (noahheller) [inline src/main/java/frc/robot/Constants.java]: Should be all CAPS to follow convention

**Review by bengold22 — APPROVED** [review]

**Pull request merged**